src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListServiceImpl.java

```diff
@@ -158,4 +337,289 @@ public void deleteGermplasmListFolder(final Integer folderId) {
+
+		this.addListDataProperties(propertyName.getName(),
+			valueSupplier,
+			listDataIndexedByGid);
+	}
+
+	private <T> void addListDataProperties(final String propertyName, final Supplier<Map<Integer, T>> valuesSupplier, final Map<Integer, GermplasmListData> listDataIndexedByGid) {
+		valuesSupplier.get()
+			.entrySet()
+			.forEach(entrySet -> this.addListDataProperty(listDataIndexedByGid.get(entrySet.getKey()),
+				propertyName,
+				Objects.isNull(entrySet.getValue()) ? null : entrySet.getValue().toString()));
+	}
+
+	private <T> void addListDataProperty(final GermplasmListData listData, final String propertyName, final String value) {
+		final ListDataProperty listDataProperty = new ListDataProperty(
+			listData,
+			propertyName,
+			value
+		);
+		this.daoFactory.getListDataPropertyDAO().save(listDataProperty);
+	}
+
+	private void addCrossFemaleDataPropertyValues(final GermplasmListDataPropertyName propertyName, final Map<Integer, GermplasmListData> listDataIndexedByGid) {
+		final Integer level = this.crossExpansionProperties.getCropGenerationLevel(this.pedigreeService.getCropName());
+		final Table<Integer, String, Optional<Germplasm>>
+			table = this.pedigreeDataManager.generatePedigreeTable(listDataIndexedByGid.keySet(), level, false);
+
+		listDataIndexedByGid.keySet().forEach(gid -> {
+			final Optional<Germplasm> femaleParent = table.get(gid, GermplasmListDataPropertyName.FGID.getName());
+			final String value;
+			if (propertyName == GermplasmListDataPropertyName.FGID) {
+				value = this.getGermplasmGid(femaleParent);
+			} else if (propertyName == GermplasmListDataPropertyName.CROSS_FEMALE_PREFERRED_NAME) {
+				value = this.getGermplasmPreferredName(femaleParent);
+			} else {
+				value = "-";
+			}
+
+			this.addListDataProperty(listDataIndexedByGid.get(gid), propertyName.getName(), value);
+		});
+	}
+
+	private void addCrossMaleDataPropertyValues(final GermplasmListDataPropertyName propertyName, final Map<Integer, GermplasmListData> listDataIndexedByGid) {
+
+		final Integer level = this.crossExpansionProperties.getCropGenerationLevel(this.pedigreeService.getCropName());
+		final Table<Integer, String, Optional<Germplasm>> table = this.pedigreeDataManager.generatePedigreeTable(listDataIndexedByGid.keySet(), level, false);
+
+		listDataIndexedByGid.keySet().forEach(gid -> {
+			final Optional<Germplasm> maleParent = table.get(gid, GermplasmListDataPropertyName.MGID.getName());
+			final String value;
+			if (propertyName == GermplasmListDataPropertyName.MGID) {
+				value = this.getGermplasmGid(maleParent);
+			} else if (propertyName == GermplasmListDataPropertyName.CROSS_MALE_PREFERRED_NAME) {
+				value = this.getGermplasmPreferredName(maleParent);
+			} else {
+				value = "-";
+			}
+			this.addListDataProperty(listDataIndexedByGid.get(gid), propertyName.getName(), value);
+		});
+	}
+
+	private String getGermplasmGid(final Optional<Germplasm> germplasm) {
```

Comment:
how can this be possible? Gid should always be present if the germplasm is already saved @lik-83

Reply:
dismiss this one, I just saw that it is used for the parents